src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -268,52 +286,288 @@ internal BoundExpression ConvertPatternExpression(TypeSymbol inputType, CSharpSy
+            ImmutableArray<(Symbol property, BoundPattern pattern)> propertiesOpt = BindPropertySubpattern(node.PropertySubpattern, declType, diagnostics, ref hasErrors);
+            BindPatternDesignation(node, node.Designation, declType, typeSyntax, diagnostics, ref hasErrors, out var variableSymbol, out var variableAccess);
+            return new BoundRecursivePattern(
+                syntax: node, declaredType: boundDeclType, inputType: declType, deconstructMethodOpt: null,
+                deconstruction: default, propertiesOpt: propertiesOpt, variable: variableSymbol, variableAccess: variableAccess, hasErrors: hasErrors);
+        }
+
+        ImmutableArray<(Symbol property, BoundPattern pattern)> BindPropertySubpattern(
+            PropertySubpatternSyntax node,
+            TypeSymbol inputType,
+            DiagnosticBag diagnostics,
+            ref bool hasErrors)
+        {
+            var builder = ArrayBuilder<(Symbol property, BoundPattern pattern)>.GetInstance();
+            foreach (var p in node.SubPatterns)
+            {
+                var name = p.NameColon?.Name;
+                var pattern = p.Pattern;
+                Symbol property = null;
+                TypeSymbol propertyType;
+                if (name == null)
+                {
+                    if (!hasErrors)
+                    {
+                        diagnostics.Add(ErrorCode.ERR_PropertyPatternNameMissing, pattern.Location, pattern);
+                    }
+
+                    propertyType = CreateErrorType();
+                    hasErrors = true;
+                }
+                else
+                {
+                    property = LookupPropertyForPattern(inputType, name, out propertyType, ref hasErrors, diagnostics);
+                }
+
+                var boundPattern = BindPattern(pattern, propertyType, hasErrors, diagnostics);
+                builder.Add((property, boundPattern));
+            }
+
+            return builder.ToImmutableAndFree();
+        }
+
+        private Symbol LookupPropertyForPattern(TypeSymbol inputType, IdentifierNameSyntax name, out TypeSymbol propertyType, ref bool hasErrors, DiagnosticBag diagnostics)
+        {
+            var symbol = BindPropertyPatternMember(inputType, name, ref hasErrors, diagnostics);
+
+            if (inputType.IsErrorType() || hasErrors)
+            {
+                propertyType = CreateErrorType();
+                return null;
             }
+
+            propertyType = symbol.GetTypeOrReturnType();
+            return symbol;
         }
 
+        private Symbol BindPropertyPatternMember(
+            TypeSymbol inputType,
+            IdentifierNameSyntax memberName,
+            ref bool hasErrors,
+            DiagnosticBag diagnostics)
+        {
+            // TODO: consider refactoring out common code with BindObjectInitializerMember
```

Comment:
>TODO [](start = 15, length = 4)

PROTOTYPE?

Reply:
Resolved in [https://github.com/dotnet/roslyn/pull/23357](https://github.com/dotnet/roslyn/pull/23357)

---
In reply to: [156519222](https://github.com/dotnet/roslyn/pull/23209#discussion_r156519222) [](ancestors = 156519222)